tests/test_sleep_block.py

```diff
@@ -0,0 +1,123 @@
+        blk.process_signals(signals)
+        self.assert_num_signals_notified(0)
+        e.wait()
+        end = time.time()
+        self.assert_num_signals_notified(2)
+        self.assertTrue(0.1 < end - start < 0.5)
+        blk.stop()
+
+    def test_persist_save(self):
+        blk = Sleep()
+        self.configure_block(blk, {
+            'log_level': 'DEBUG',
+            'interval': {'seconds': 10}
+        })
+        blk.persistence.store = MagicMock()
+        blk.persistence.save = MagicMock()
+        blk.start()
+        signals_a = [Signal({'name': 'signal1'}),
+                     Signal({'name': 'signal2'})]
+        signals_b = [Signal({'name': 'signal3'}),
+                     Signal({'name': 'signal4'})]
+
+        def assert_persistence(store_count, save_count):
+            self.assertEqual(blk.persistence.store.call_count, store_count)
+            self.assertEqual(blk.persistence.save.call_count, save_count)
+            self.assertEqual(blk.persistence.store.call_args[0][0],
+                            'signals')
+            self.assertEqual(blk.persistence.store.call_args[0][1],
+                            blk._signals)
+
+        # process some signals
+        blk.process_signals(signals_a)
+        assert_persistence(1, 0)
+        # process some more signals
+        blk.process_signals(signals_b)
+        assert_persistence(2, 0)
+        # stop the block and save persistence
+        blk.stop()
+        assert_persistence(3, 1)
+
+        # the block should have stopped before any signals were notified
+        self.assert_num_signals_notified(0)
+        # assert that two groups of signals were saved
+        self.assertEqual(len(blk._signals), 2)
+        # assert that each group has two signals
+        self.assertEqual(len(blk._signals[0][1]), 2)
+        self.assertEqual(len(blk._signals[1][1]), 2)
+        # assert that the second set of signals has a later time than the first
+        self.assertTrue(blk._signals[0][0] < blk._signals[1][0])
+
+    @patch(Sleep.__module__ + '._time')
+    def test_persist_load(self, mock_time):
+        e = Event()
+        blk = SleepEvent(e)
+        signals1 = [Signal()]
+        signals2 = [Signal(), Signal()]
+        signals = [(0, signals1), (1, signals2)]
+        with patch('nio.modules.persistence.default.Persistence.load') as load:
+            load.return_value = signals
+            self.configure_block(blk, {
+                'log_level': 'DEBUG',
+                'interval': {'seconds': 1}
+            })
+        mock_time.return_value = 0.9
+        blk.start()
+        # check that all signals were loaded but only two signals were put
+        # pack into _signals
+        self.assertEqual(blk._load_signals, signals)
+        self.assertEqual(len(blk._signals), 1)
+        self.assertEqual(len(blk._signals[0][1]), 2)
+        self.assert_num_signals_notified(0)
+        e.wait()
+        self.assert_num_signals_notified(2)
```

Comment:
Could be worth adding a `blk.stop` after and making sure they don't get persisted again